src/client.js

```diff
@@ -0,0 +1,40 @@
+/**
+* @fileoverview Bard client library for shared DSP client-side analytics
+*/
+
+import {_} from './utils'
+import Config from './config'
+
+/**
+ * Removes full "referrer" in cases required by DSP
+ *
+ * If navigating from e.g.:
+ *
+ *   - A Terra workspace to another Terra page, or
+ *   - An SCP study to another SCP page
+ *
+ * The "referrer" includes workspace name or study name.  DSP considers
+ * these private data not suitable to log in Mixpanel.  So here we strip
+ * referrer if the referring domain is for a DSP product.
+
+ * Full referrer is still useful for external domains (and can be truncated
+ * or omitted by them as they desire), so we retain it for those cases.
+ *
+ * @param {Object} defaultProps
+ */
+function filterReferrer(defaultProps) {
+  const referringDomain = defaultProps['referring_domain'];
+  if (Config.DSP_WEB_UI_ROOTS.includes(referringDomain)) {
+    delete defaultProps['referrer']
```

Comment:
Can we just truncate this to at least preserve analytics on cross-DSP referrals?
```suggestion
    defaultProps['referrer'] = new URL(referringDomain).hostname
```

Reply:
Nice!  This abides by privacy constraints (don't reveal workspace / study name), and while it's redundant with `referring_domain` in this case, it avoids the need to stitch together `referrer` and `referring_domain` in Mixpanel queries.

This might be even more useful if we include `appFullPath` in `referrer` here, but that's a more involved idea for later.